File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlExistsExpression.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression } from './ISqlExpression';

export class SqlExistsExpression implements ISqlExpression {
    SqlType: SqlType = SqlType.ExistsExpression;
    Span: TextSpan = new TextSpan();
    Query!: ISqlExpression;

    Accept(visitor: any): void {
        // 簡單實作
    }

    ToSql(): string {
        return `EXISTS (${this.Query.ToSql()})`;
    }
} 

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlExistsExpression.cs
namespace T1.SqlSharp.Expressions;

public class SqlExistsExpression : ISqlExpression
{
    public SqlType SqlType { get; set; } = SqlType.ExistsExpression;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_ExistsExpression(this);
    }

    public required ISqlExpression Query { get; set; }
    public string ToSql()
    {
        return $"EXISTS ({Query.ToSql()})";
    }
}